types/promise-hash/index.d.ts

```diff
@@ -0,0 +1,16 @@
+// Type definitions for promise-hash 1.3
+// Project: https://github.com/mtimofiiv/promise-hash
+// Definitions by: Michael Shafir <https://github.com/mshafir>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export = hash;
+
+type PromiseHash<T> = { [P in keyof T]: PromiseLike<T[P]> | T[P] };
```

Comment:
I was thinking along the lines of [this](https://www.typescriptlang.org/play/index.html?ssl=1&ssc=1&pln=22&pc=57#code/C4TwDgpgBACgTgewLYEsDOEASBDNALKAXigB4AVAPgAoxFUMd8AuKAbygG0YoUA7KANYQQCAGZQyAXRbxk6CABkUQ8l0kUoAHwlqoAXwCURDbPoRyFANwAoawBMIAYwA22ONADmzhACNsztmsoYJ5eYAg4UWxHaFN5AGEEXjRgOABXR2AEOECQvKg8XDwZOnlGPBs8vWtq+ydXdyhHJJSCopK5BiKba2bk4Ch3NDTnAeJC-CpWIJDsDrMAOiGEZwA3CCoAVgMAGhngn3n5JYg0FfWqACJwlMvd-aaWCbwph7y7I4wTs7WN1LSIPd8iEIJ8IN9zhtLtgPNg+HcHoYagYeqBIFAxrBShgSOw5lBeGkkD4IpYoIcoCk4HwPGTHCx2B9yQgVhBsLwyaDKakaWS9HyrNY0dAAB5EDHgCBiQanEYDCAi8K8OxoDFQAD8GPS0BYUWcGDJAHpDVqAULJVAQOL5YqIMrVcLpUM5RrTTqoHqDVBjW6gA). I think it's just a matter of preference, so I'm happy to merge either one.

Reply:
Cool! I had no idea you could put the generic <T> in front of the anonymous function like that. I'll make the update.